pkg/client/message.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+type Action string
+
+const (
+	// Аутентификация и ассоциация
+	ActionAssociate        Action = "associate"          // Ассоциация с браузерным расширением
+	ActionTestAssociate    Action = "test-associate"     // Проверка статуса ассоциации
+	ActionChangePublicKeys Action = "change-public-keys" // Обмен публичными ключами
+
+	// Операции с базой данных
+	ActionGetDatabaseHash    Action = "get-databasehash"     // Получение хеша базы данных
+	ActionLockDatabase       Action = "lock-database"        // Блокировка базы данных
+	ActionGetDatabaseGroups  Action = "get-database-groups"  // Получение списка групп
+	ActionCreateNewGroup     Action = "create-new-group"     // Создание новой группы
+	ActionGetDatabaseEntries Action = "get-database-entries" // Получение всех записей
+
+	// Управление логинами и паролями
+	ActionGetLogins        Action = "get-logins"        // Получение учетных данных
+	ActionSetLogin         Action = "set-login"         // Сохранение/обновление учетных данных
+	ActionDeleteEntry      Action = "delete-entry"      // Удаление записи
+	ActionGeneratePassword Action = "generate-password" // Генерация нового пароля
+	ActionGetTotp          Action = "get-totp"          // Получение TOTP кода
+
+	// Автоматизация
+	ActionRequestAutotype Action = "request-autotype" // Запуск автоввода
+
+	// Операции с Passkeys (если скомпилировано с WITH_XC_BROWSER_PASSKEYS)
+	ActionPasskeysGet      Action = "passkeys-get"      // Получение учетных данных passkey
+	ActionPasskeysRegister Action = "passkeys-register" // Регистрация нового passkey
+)
+
+type ErrorCode int
+
+const (
+	ErrorKeePassIncorrectAction ErrorCode = iota
+	ErrorKeePassEmptyMessageReceived
+	ErrorKeePassClientPublicKeyNotReceived
+	ErrorKeePassCannotDecryptMessage
+	ErrorKeePassDatabaseNotOpened
+	ErrorKeePassDatabaseHashNotReceived
+	ErrorKeePassAssociationFailed
+	ErrorKeePassActionCancelledOrDenied
+	ErrorKeePassEncryptionKeyUnrecognized
+	ErrorKeePassNoUrlProvided
+	ErrorKeePassNoLoginsFound
+	ErrorKeePassNoGroupsFound
+	ErrorKeePassCannotCreateNewGroup
+	ErrorKeePassNoValidUuidProvided
+	ErrorKeePassAccessToAllEntriesDenied
+	ErrorPasskeysEmptyPublicKey
+	ErrorPasskeysInvalidUrlProvided
+)
+
+type MessageKeys struct {
+	ID  string `json:"id"`
+	Key string `json:"key"`
+}
+
+type Message struct {
```

Comment:
AFAICS messages serves just a wrapper structure for various kinds of messages and fields could differ from one to another, so in current implementation it's up to consumer of the structure to know what fields are required for every particular message and pass them well-formed. Probably we could create constructors for them to move this logic out of consumer's code?